docs/reference/analysis/synonyms.asciidoc

```diff
@@ -0,0 +1,261 @@
+[[analysis-synonyms]]
+== Synonyms
+
+Synonyms are words or phrases that have the same or similar meaning.
+They are an important aspect of search, as they can improve the search experience and the relevancy of search results.
+
+Synonyms allow you to:
+
+* Improve search relevance: Synonyms enable users to find relevant documents even if they use different terms to express the same concept.
+* Increase search scope: Using synonyms ensures a wider range of possible terms. Search results are not limited to the specific terms included in your documents, and variations of relevant terms are included in search results.
+* Domain-specific vocabulary: Certain domains use specific terms that users are not familiar with. Using synonyms allow search to become more user-friendly, as users can search terms they are familiar with and search results will not be limited to the specific domain vocabulary.
+* Handling common misspellings and typos: Synonyms can be used to make sure that frequent or specific misspellings are correctly handled.
+
+Synonyms are grouped together using *synonyms sets*.
+You can have as many synonyms sets as you need.
+
+In order to use synonyms sets in {es}, you need to:
+
+* <<synonyms-define-synonyms>>
+* <<synonyms-store-synonyms>>
+* <<synonyms-synonym-analyzers>>
+
+[discrete]
+[[synonyms-define-synonyms]]
+=== Define synonyms sets
+
+Synonyms in a synonyms set are defined using *synonym rules*.
+Each synonym rule contains words that are synonyms.
+
+You can use two formats to define synonym rules: Solr and WordNet.
+
+[discrete]
+[[synonyms-define-synonyms-solr]]
+==== Solr format
+
+This format uses two different definitions:
+
+* Equivalent synonyms: Define groups of words that are equivalent. Words are separated by commas. Example:
+[source,synonyms]
+----
+ipod, i-pod, i pod
+computer, pc, laptop
+----
+* Explicit mappings: Matches a group of words to other words. Words on the left hand side of the rule definition are expanded into all the possibilities described on the right hand side. Example:
+
+[source,synonyms]
+----
+personal computer => pc
+sea biscuit, sea biscit => seabiscuit
+----
+
+[discrete]
+[[synonyms-define-synonyms-wordnet]]
+==== WordNet format
+
+https://wordnet.princeton.edu/[WordNet] defines synonyms sets spanning multiple lines. Each line contains the following information:
```

Comment:
I think better to not provide any links, but in case you want to provide we need to provide a link to the wordnet format: https://wordnet.princeton.edu/documentation/prologdb5wn

Reply:
I haven't found a direct link to the format used here. Checking the link you provided, I don't see that it is a direct description of the format either 😢 

I'm happy to remove the link, but I think it provides an initial link where users can investigate.